aineko/extras/connectors/rest_poller.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Extra module for connecting to a REST endpoint."""
+
+import json
+import time
+from typing import Any, Dict, Optional
+
+import requests
+from pydantic import BaseModel, field_validator
+
+from aineko import AbstractNode
+
+
+class ParamsRESTPoller(BaseModel):
+    """Connector params for REST model."""
+
+    timeout: int = 10
+    url: str
+    headers: Optional[Dict[str, Any]] = None
+    data: Optional[Dict[str, Any]] = None
+    poll_interval: int = 5
+    poll_throttle: float = 0.1
+    max_retries: int = 30
+    metadata: Optional[Dict[str, Any]] = None
+    retry_sleep: float = 5
+
+    @field_validator("url")
+    @classmethod
+    def supported_url(cls, url: str) -> str:  # pylint: disable=no-self-argument
```

Comment:
```suggestion
    def supported_url(cls, url: str) -> str:
```
Not needed when adding the `@classmethod` decorator.

Reply:
got it, fixed